python/feathub/processors/flink/table_builder/metric_store_utils.py

```diff
@@ -0,0 +1,151 @@
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import collections
+import json
+import typing
+
+from feathub.common.exceptions import FeathubException
+from feathub.feature_tables.sinks.prometheus_sink import PrometheusSink
+from feathub.feature_tables.sinks.sink import Sink
+from feathub.feature_views.derived_feature_view import DerivedFeatureView
+from feathub.feature_views.feature import Feature
+from feathub.feature_views.sliding_feature_view import SlidingFeatureView
+from feathub.feature_views.transforms.sliding_window_transform import (
+    SlidingWindowTransform,
+)
+from feathub.metric_stores.metric import Metric, Count, Ratio
+from feathub.metric_stores.metric_store import MetricStore
+from feathub.metric_stores.prometheus_metric_store import PrometheusMetricStore
+from feathub.processors.flink.table_builder.source_sink_utils_common import (
+    generate_random_name,
+)
+from feathub.processors.materialization_descriptor import MaterializationDescriptor
+from feathub.table.table_descriptor import TableDescriptor
+
+
+def _get_metric_name(metric: Metric, metric_store: MetricStore) -> str:
+    return f"feathub_{metric_store.namespace}_feature_{metric.metric_type}"
+
+
+def _get_metric_tags(metric: Metric, feature: Feature) -> typing.OrderedDict[str, str]:
+    tags = collections.OrderedDict(
+        [
+            ("feature_name", feature.name),
+            ("window_time", str(metric.window_size)),
+        ]
+    )
+    if isinstance(metric, Count) or isinstance(metric, Ratio):
+        tags["filter_expr"] = metric.filter_expr
+    return tags
+
+
+def _get_prometheus_grouping_keys(data_sink: Sink) -> typing.Dict[str, str]:
+    return {"table_name": data_sink.name}
+
+
+# TODO: support treating zero window as infinite window in SlidingWindowTransform
+def _get_metrics_view(
+    features_desc: TableDescriptor, metric_store: MetricStore
+) -> TableDescriptor:
+    if not isinstance(metric_store, PrometheusMetricStore):
+        raise FeathubException(f"Unsupported metric store type {type(metric_store)}.")
+
+    metric_features = []
+    post_process_metric_features = []
+    for feature in features_desc.get_output_features():
+        for metric in feature.metrics:
+            metric_tags = _get_metric_tags(metric, feature)
+            metric_tags["metric_name"] = _get_metric_name(metric, metric_store)
+            feature_name = generate_random_name("feathub_metric")
+            if isinstance(metric, Count):
+                metric_features.append(
+                    Feature(
+                        name=feature_name,
+                        transform=SlidingWindowTransform(
```

Comment:
Would it be more extensible to move the logic of converting metric definition into feature definition into the corresponding metric subclasses?